annotation/core/views.py

```diff
@@ -202,18 +202,39 @@ def save(request):
     text = int(request.POST['text'])
     name = request.POST['name']
     boundary = int(request.POST['boundary'])
-    revision = request.POST['revision']
     points = request.POST['points']
     attention_check = request.POST['attention_check']
 
+    grammar = request.POST['grammar'] == 'true'
+    repetition = request.POST['repetition'] == 'true'
+    irrelevant = request.POST['irrelevant'] == 'true'
+    contradicts_sentence = request.POST['contradicts_sentence'] == 'true'
+    contradicts_knowledge = request.POST['contradicts_knowledge'] == 'true'
+    common_sense = request.POST['common_sense'] == 'true'
+    coreference = request.POST['coreference'] == 'true'
+    generic = request.POST['generic'] == 'true'
+    other_reason = request.POST['other_reason']
+
     annotation = Annotation.objects.create(
         annotator=request.user,
         generation=Generation.objects.get(pk=text),
         boundary=boundary,
-        revision=revision,
         points=points,
         attention_check=attention_check
     )
+    
+    if grammar: annotation.reason.add(FeedbackOption.objects.get(shortname="grammar"))
+    if repetition: annotation.reason.add(FeedbackOption.objects.get(shortname="repetition"))
+    if irrelevant: annotation.reason.add(FeedbackOption.objects.get(shortname="irrelevant"))
+    if contradicts_sentence: annotation.reason.add(FeedbackOption.objects.get(shortname="contradicts_sentence"))
+    if contradicts_knowledge: annotation.reason.add(FeedbackOption.objects.get(shortname="contradicts_knowledge"))
+    if common_sense: annotation.reason.add(FeedbackOption.objects.get(shortname="common_sense"))
+    if coreference: annotation.reason.add(FeedbackOption.objects.get(shortname="coreference"))
+    if generic: annotation.reason.add(FeedbackOption.objects.get(shortname="generic"))
+
+    if other_reason:
+        new_reason = FeedbackOption.objects.create(shortname = "other_reason", description = other_reason)
```

Comment:
Since we are going to make shortname a primary key, that means it needs to be unique. One easy way to do that is with:
```
new_reason = FeedbackOption.objects.create(shortname = str(hash(other_reason)), description = other_reason)
```

Reply:
You'll also want to set is_default to False, so it should actually be:
```
new_reason = FeedbackOption.objects.create(shortname = str(hash(other_reason)), description = other_reason, is_default = False)
```

Reply:
fixed!